planner/core/rule_decorrelate.go

```diff
@@ -192,7 +192,7 @@ func (s *decorrelateSolver) optimize(ctx context.Context, p LogicalPlan) (Logica
 					switch expr := aggFunc.Args[0].(type) {
 					case *expression.Column:
 						if idx := apply.schema.ColumnIndex(expr); idx != -1 {
-							desc, err := aggregation.NewAggFuncDesc(agg.ctx, agg.AggFuncs[i].Name, []expression.Expression{apply.schema.Columns[idx]}, false)
+							desc, err := aggregation.NewAggFuncDesc(agg.ctx, agg.AggFuncs[i].Name, []expression.Expression{apply.schema.Columns[idx]}, agg.AggFuncs[i].HasDistinct)
```

Comment:
I found that many usages use `false`, does they need to change too?

Reply:
Need consider situation case by case.  In this case, it is an aggFuncs rewrite logic, so we should keep the property.

Reply:
The AggFuncs in other usage is only FirstRow,Min/Max,Count.   FirstRow,Min/Max don't need to consider distinct.
There are only four usages of the function by Count.  It seems that their usage is correct.